renga_deployer/engines.py

```diff
@@ -260,12 +280,13 @@ def launch(self, execution, engine=None, **kwargs):
 
             # if using an ingress, need to make an additional object
             if current_app.config.get(
-                    'DEPLOYER_K8S_USE_INGRESS'):  # pragma no cover
+                    'DEPLOYER_K8S_USE_INGRESS'):
                 beta_api = self._kubernetes.client.ExtensionsV1beta1Api()
                 ingress = beta_api.create_namespaced_ingress(
                     namespace,
                     self._k8s_ingress_template(uid, service.metadata.name,
-                                               context.spec['ports'][0]))
+                                               context.spec['ports'][0],
+                                               execution.id))
```

Comment:
I would propose, that we pass only the `execution` to `_k8s_ingress_template` and the `execution.id` and `execution.context` is extracted inside the method.